GVRf/Framework/src/org/gearvrf/GVRContext.java

```diff
@@ -489,6 +493,22 @@ public GVRBitmapTexture loadTexture(GVRAndroidResource resource) {
         return bitmap == null ? null : new GVRBitmapTexture(this, bitmap);
     }
 
+    public GVRCubemapTexture loadCubemapTexture(
+            List<GVRAndroidResource> resourceList) {
+
+        assertGLThread();
+
+        List<Bitmap> bitmapList = new ArrayList<Bitmap>(6);
+        for (GVRAndroidResource resource : resourceList) {
+            bitmapList.add(GVRAsynchronousResourceLoader.decodeStream(
+                    resource.getStream(), false));
+            resource.closeStream();
+        }
+
+        return resourceList.size() != 6 ? null : new GVRCubemapTexture(this,
+                bitmapList);
```

Comment:
I think the size check should be at the beginning other wise it may be a performance waste when the texture is not valid.